frontend/src/components/DeckyState.tsx

```diff
@@ -128,9 +128,17 @@ interface DeckyStateContext extends PublicDeckyState {
   closeActivePlugin(): void;
 }
 
-const DeckyStateContext = createContext<DeckyStateContext>(null as any);
+const DeckyStateContext = createContext<DeckyStateContext | null>(null);
 
-export const useDeckyState = () => useContext(DeckyStateContext);
+export const useDeckyState = () => {
+  const deckyState = useContext(DeckyStateContext);
+
+  if (deckyState === null) {
+    throw new Error('useDeckyState needs a parent DeckyStateContext');
+  }
+
+  return deckyState;
+};
```

Comment:
Note for reviewers: this tripped me up since the types assume this will never be `null`, which might not be true if `DeckyStateContext` is missing.

Opted to add this runtime check for some clarity on the error instead of failing later down the line.